CONTRIBUTING.md

```diff
@@ -1,39 +1,78 @@
-# prometheus-scrape-target
+# Contributing
+
+## Overview
+
+This documents explains the processes and practices recommended for
+contributing enhancements to the Prometheus Scrape Target charm.
+
+- Generally, before developing enhancements to this charm, you should consider
+  [opening an issue ](https://github.com/canonical/prometheus-scrape-target-k8s-operator)
+  explaining your use case.
+- If you would like to chat with us about your use-cases or proposed
+  implementation, you can reach us at
+  [Canonical Mattermost public channel](https://chat.charmhub.io/charmhub/channels/charm-dev)
+  or [Discourse](https://discourse.charmhub.io/).
+  The primary author of this charm is available on the Mattermost channel as
+  `@balbir-thomas`.
+- Familiarising yourself with the
+  [Charmed Operator Framework](https://juju.is/docs/sdk)
+  library will help you a lot when working on new features or bug fixes.
+- All enhancements require review before being merged. Code review
+  typically examines
+  + code quality
+  + test coverage
+  + user experience for Juju administrators
+- Please help us out in ensuring easy to review branches by rebasing
+  your pull request branch onto the `main` branch. This also avoids
+  merge commits and creates a linear Git commit history.
```

Comment:
This last part is true regardless since we only allow squash merges. But no biggie.